rfc/0075-faster-failover-and-configuration-push.md

```diff
@@ -0,0 +1,315 @@
+        +---------------+---------------+---------------+---------------+
+      32| 0x00          | 0x00          | 0x00          | 0x00          |
+        +---------------+---------------+---------------+---------------+
+      36| 0x08          | 0x07          | 0x06          | 0x05          |
+        +---------------+---------------+---------------+---------------+
+      40| 0x04          | 0x03          | 0x02          | 0x01          |
+        +---------------+---------------+---------------+---------------+
+    CLUSTERMAP_CHANGE_NOTIFICATION command
+    Field        (offset) (value)
+    Magic        (0)    : 0x82 (server request, kv_engine -> SDK)
+    Opcode       (1)    : 0x01
+    Key length   (2,3)  : 0x0000
+    Extra length (4)    : 0x10 (two int64_t fields in extras)
+    Data type    (5)    : 0x00 (RAW)
+    Vbucket      (6,7)  : 0x0000
+    Total body   (8-11) : 0x00000010 (16 bytes)
+    Opaque       (12-15): 0x00000000
+    CAS          (16-23): 0x0000000000000000
+    Epoch        (24-31): 0x0000000000000042 (66 in base-10)
+    Revision     (32-39): 0x0102030405060708 (72623859790382856 in base-10)
+
+So note that magic is `ServerRequest` (`0x82`), that is enabled by `Duplex` (`0x0c`) HELLO flag. Also note that just
+like in regular cluster configuration notification, epoch and revision fields are sent as extras.
+
+Note that the magic value for this notification is `ServerRequest` (`0x82`), which is enabled by the `Duplex` (`0x0c`)
+HELLO flag. Additionally, similar to the regular cluster configuration notification, the epoch and revision fields are
+sent as extras.
+
+Once the brief cluster configuration notification is received, it is up to the SDK to decide whether to send a
+`GetClusterConfig` (`0xb5`) request to retrieve the actual configuration body.
+
+In essence, the `ClustermapChangeNotificationBrief` feature only saves network traffic. If
+`DedupeNotMyVbucketClustermap` is not enabled, the number of notifications will be the same as before. However, this
+feature can still be used as a building block to implement a debouncing mechanism. When properly configured, it can help
+reduce the number of requests. Further details on this topic will be covered in the "Library Changes" section.
+
+## Library Changes
+
+### Configuration Push
+
+The previously mentioned `ClustermapChangeNotificationBrief` feature enables the SDK to subscribe all connections for
+configuration updates. These notifications are lightweight and can be deduplicated by the server when the
+`DedupeNotMyVbucketClustermap` option is negotiated.
+
+#### Mixed Clusters
+
+In clusters where there is a mix of nodes with older server versions, meaning that some nodes do not acknowledge
+`ClustermapChangeNotificationBrief`, the respective connection should notify the configuration monitor about its lack of
+support for configuration pushes from the server. As a result, the monitor should utilize the old polling mechanism for
```

Comment:
Does this mean that the SDK will have to continue sending CCCP requests but only to a subset of nodes in a mixed cluster?

Reply:
Yes, it will be safe assumption, but it is open for discussion.

Reply:
I have some concerns over the complexity this is going to add, something will have to monitor how many nodes in the cluster are still using the old mechanism - at least in Go anyway. It seems like it'd be a lot simpler if there was a cluster capability or something that the SDK could use as a signal to switch over.

Reply:
I also agree that this seems quite a bit more complicated than necessary.  Perhaps a simpler logic might be that if at least 3 nodes (or all the nodes in the cluster, if less than 3) acknowledge config push, we disable polling, but otherwise use polling.

Reply:
that still requires counting nodes. Do you mean that we should still use polling and push if there are 5 nodes in cluster, where only 2 nodes support push?

I see it like we still have list of nodes which are used for round-robin, we just need to remove from this list addresses, that ack Push Hello flag.

Reply:
Updated mixed mode case